### PR TITLE
Fix forceUpdate after react 16.8.1 upgrade

### PR DESCRIPTION
**Pull request by brunolemos**
Fix https://github.com/react-spring/react-spring/issues/490

Fix https://github.com/facebook/react/issues/14781

### CODE REVIEW

File: src/renderprops/hooks/useKeyframes.js [hunk 1]
```diff
@@ -51,11 +52,14 @@ const useKeyframesImpl = useImpl => (props, initialProps = null) => (
...
-    setAnimation(states[state])
-  }, [state])
+  React.useEffect(
+    () => {
+      shouldForceUpdateRef.current && forceUpdate()
+      shouldForceUpdateRef.current = false
+      setAnimation(states[state])
+    },
+    [state]
+  )
```
Comment (brunolemos) [inline src/renderprops/hooks/useKeyframes.js]: I didnt make this change

File: src/renderprops/hooks/useSprings.js [hunk 2]
```diff
@@ -17,31 +17,35 @@ export const useSpringsImpl = (type = 'default', trail = false) => (
...
+        instances.push(
+          type === 'keyframe'
+            ? new KeyframeController(initProps)
+            : new Ctrl(initProps)
+        )
+      }
+      return instances
+    },
+    [length]
+  )
```
Comment (brunolemos) [inline src/renderprops/hooks/useSprings.js]: I didnt make this change

File: src/renderprops/hooks/useTransition.js [hunk 3]
```diff
@@ -179,61 +180,67 @@ export function useTransition(props) {
...
+      state.current = {
+        ...state.current,
+        transitions,
+        prevProps: props,
+        first: true,
+        ...rest,
+      }
+    },
+    [items, mapKeys(items, _currentKeys).join('')]
+  )
```
Comment (brunolemos) [inline src/renderprops/hooks/useTransition.js]: Didnt make this change

**Comment by drcmda** [comment]
oh no, sorry, i've seen this to late :( already merged a fix

**Comment by brunolemos** [comment]
https://github.com/react-spring/react-spring/commit/8dc0bc7fa3c53e42558853b80c9353538ad30a70

**Pull request closed without merging**